### DISCUSSION TITLE
Helper function for sending InSim buttons

### DISCUSSION THREAD

**Discussion by mkapal** [simbroadcasts/node-insim#60] [Ideas] [answered]
### Problem
- If you want to create a button with a type-in dialog including a caption, you must send the caption in the `Text` property delimited by zero-bytes: `'\0Caption\0Text'`.

### Current API

```ts
inSim.send(new IS_BTN({
  ReqI: 1,
  L: 60,
  T: 4,
  W: 20,
  H: 5,
  TypeIn: 20,
  Text: '\0Caption\0Text',
}))
```

### Proposed API

```ts
inSim.sendButton({
  requestId: 1,
  left: 60,
  top: 4,
  width: 20,
  height: 5,
  text: 'Text',
  typeIn: 20,
  caption: 'Caption',
})
```

**Comment by Degats** [comment, marked as answer]
From experience, it's very useful to have abstractions like these for some specific packets, especially buttons as the Text field has some complex behaviour.

**Comment by theangryangel** [comment]
Something you may also want to have in the back of your head, at *some point* in insim.rs I had a `send_and_wait_for_response` style method where it would automatically set the `reqi` on a given packet (ensuring it was unique on a per-insim-connection basis), and then wait for a response where the `reqi` was the same, and then return the packet.

I dropped it, with the plan to bring it back (eventually) - but it is something I've needed at some point in most insim things I've worked on.

**Comment by mkapal** [comment]
@theangryangel Thanks, that's a good idea, I will consider it for the future.

**Comment by mkapal** [comment]
I've split this issue into two, created a new one for messages: #54
